### PR TITLE
All-Pairs Shortest Paths

### PR DESCRIPTION
**Pull request by armcknight**
Just claiming this algorithm! Plan to do the Floyd-Warshall version, at least at first.

### CODE REVIEW

**Comment by armcknight** [comment]
Modified the AdjacencyMatrix.swift unit defining the Graph data structure so I could use it in this.

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 language: objective-c
-osx_image: xcode7.2
+osx_image: xcode7.3
```
Comment (armcknight) [inline .travis.yml]: Looks like all the tests in `.travis.yml` still pass with this change!

File: .travis.yml [hunk 2]
```diff
@@ -23,4 +23,5 @@ script:
 - xcodebuild test -project ./Shell\ Sort/Tests/Tests.xcodeproj -scheme Tests
 - xcodebuild test -project ./Stack/Tests/Tests.xcodeproj -scheme Tests
 - xcodebuild test -project ./Longest\ Common\ Subsequence/Tests/Tests.xcodeproj -scheme Tests
-- xcodebuild test -project ./Bucket\ Sort/Tests/Tests.xcodeproj -scheme Tests
\ No newline at end of file
+- xcodebuild test -project ./Bucket\ Sort/Tests/Tests.xcodeproj -scheme Tests
+- xcodebuild test -project ./All-Pairs\ Shortest\ Paths/APSP/APSP.xcodeproj -scheme APSP
```
Comment (armcknight) [inline .travis.yml]: do you prescribe a particular project/target/etc structure for submissions? Related to my question in #110 about dependency managers...

**Comment by armcknight** [comment]
This should be all set for review @hollance. I'll squash when it's ready to be merged.

**Comment by hollance** [comment]
Looks good! Just to make sure we're not violating any copyrights, did you create those images yourself or did you copy them from somewhere?

**Comment by armcknight** [comment]
@hollance I made them myself :)

**Comment by hollance** [comment]
Excellent. If you squash then I'll merge it into the repo.

**Comment by armcknight** [comment]
@hollance squashed :shipit:

**Comment by hollance** [comment]
Merged! 

The README mentions an xcworkspace but I don't see that anywhere. Could you have a look at that?

**Comment by armcknight** [comment]
Ohh no, I didn't realize this before, but there is an entry for `*.xcworkspace` in the main .gitignore. Is there a reason that's in there?

**Comment by hollance** [comment]
Ah, that explains it. I updated the gitignore to allow *.xcworkspace. No idea why that was in there.

**Pull request merged**